cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs

```diff
@@ -728,76 +728,44 @@ toTxOutInAnyEra :: CardanoEra era
 toTxOutInAnyEra era (TxOutAnyEra addr' val' mDatumHash refScriptFp) = do
   addr <- hoistEither $ toAddressInAnyEra era addr'
   val <- hoistEither $ toTxOutValueInAnyEra era val'
-  (datum, refScript)
-    <- caseByronToMaryOrAlonzoEraOnwards
-         (const $ pure (TxOutDatumNone, ReferenceScriptNone))
-         (\w ->
-            caseAlonzoOnlyOrBabbageEraOnwards
-              (\wa ->
-                (,)
-                  <$> toTxAlonzoDatum (alonzoEraOnlyToAlonzoEraOnwards wa) mDatumHash
-                  <*> pure ReferenceScriptNone
-              )
-              (\wbo -> toTxDatumReferenceScriptBabbage w wbo mDatumHash refScriptFp)
-              w
-         )
-         era
+
+  datum <- caseByronToMaryOrAlonzoEraOnwards
+    (const (pure TxOutDatumNone))
+    (\wa -> toTxAlonzoDatum wa mDatumHash)
+    era
+
+  refScript <- caseByronToAlonzoOrBabbageEraOnwards
+    (const (pure ReferenceScriptNone))
+    (\wb -> getReferenceScript wb refScriptFp)
+    era
```

Comment:
This is a really good example of how you could use eons to handle each feature independently which simplifies code a lot.